packages/core/core/src/resolveOptions.js

```diff
@@ -49,14 +49,18 @@ export default async function resolveOptions(
     ])) || path.join(process.cwd(), 'index')
   );
 
+  let cacheDir =
+    // If a cacheDir is provided, resolve it relative to cwd. Otherwise,
+    // use a default directory resolved relative to the project root.
+    initialOptions.cacheDir != null
```

Comment:
This is a breaking change, but I think it makes sense. At least for the default, the cache dir shouldn't depend on where one invokes `parcel` from, but instead should be centralized at the project root.

If one is provided though, we should following Node's typical behavior of resolving relative filePaths to the pwd. If callers want to be certain, they should provide an absolute path.

This does change a few things: for example, when running integration tests, now the cache directory is at the root of the repo workspace rather than in the integration tests package.

cc @devongovett @padmaia

Reply:
Do we clean up the cache used for tests? Should we? If so, it would probably be easiest to nuke it after each test run which would be bad if we want to keep a `.parcel-cache` around for building Parcel itself. We keep saying it, but using a in memory filesystem for tests would solve this nicely.

Reply:
Nope, we don't even clean up dist directories in most cases.

I think ideally each test gets its own dist directory and cache, each in temp directories, which are cleared after each test.

Would be neat if we could use something like an overlay fs [0] (popularized by docker) to have reads of existing files go through the real filesystem, and have the remaining writes and reads be in memory.

[0] https://en.wikipedia.org/wiki/OverlayFS